src/Features/Core/Portable/IntroduceVariable/AbstractIntroduceParameterService.cs

```diff
@@ -0,0 +1,829 @@
+
+            return modifiedSolution;
+        }
+
+        /// <summary>
+        /// This method iterates through the arguments at the invocation location and maps them back
+        /// to the identifiers in the expression to create a new expression as an argument.
+        /// </summary>
+        private (TExpressionSyntax, bool) CreateNewArgumentExpression(SemanticModel invocationSemanticModel,
+            Document document, ISyntaxFactsService syntaxFacts, Dictionary<TIdentifierNameSyntax, IParameterSymbol> mappingDictionary,
+            SyntaxGenerator generator, TExpressionSyntax expression, TIdentifierNameSyntax variable,
+            SeparatedSyntaxList<SyntaxNode> invocationArguments, SeparatedSyntaxList<SyntaxNode> updatedInvocationArguments,
+            bool parameterIsNamed, CancellationToken cancellationToken)
+        {
+            if (mappingDictionary.TryGetValue(variable, out var mappedParameter))
+            {
+                var parameterMapped = false;
+                var oldNode = expression.GetCurrentNode(variable);
+                RoslynDebug.AssertNotNull(oldNode);
+                for (var i = 0; i < invocationArguments.ToArray().Length; i++)
+                {
+                    var semanticFacts = document.GetRequiredLanguageService<ISemanticFactsService>();
+                    var argumentParameter = semanticFacts.FindParameterForArgument(invocationSemanticModel, invocationArguments.ToArray()[i], cancellationToken);
+                    if (argumentParameter.Equals(mappedParameter, SymbolEqualityComparer.Default))
+                    {
+                        var updatedInvocationArgument = updatedInvocationArguments.ToArray()[i];
+                        var argumentExpression = syntaxFacts.GetExpressionOfArgument(updatedInvocationArgument);
+                        var parenthesizedArgumentExpression = generator.AddParentheses(argumentExpression, includeElasticTrivia: false);
+                        expression = expression.ReplaceNode(oldNode, parenthesizedArgumentExpression);
+                        parameterMapped = true;
+                        break;
+                    }
+                }
+
+                // This is special cased for optional parameters: if the invocation does not have an argument 
+                // corresponding to the optional parameter, then it generates an expression from the optional parameter.
+                if (mappedParameter.HasExplicitDefaultValue && !parameterMapped)
+                {
+                    parameterIsNamed = true;
+                    var generatedExpression = GenerateExpressionFromOptionalParameter(mappedParameter);
+                    var parenthesizedGeneratedExpression = generator.AddParentheses(generatedExpression, includeElasticTrivia: false);
+                    expression = expression.ReplaceNode(oldNode, parenthesizedGeneratedExpression);
+                }
+            }
+
+            return (expression, parameterIsNamed);
```

Comment:
skipping this method for now.  id ont' get it at all.

Reply:
changhed to make a lot more sense